### DISCUSSION TITLE
Create an access token (PAT) in a Heroku instant

### DISCUSSION THREAD

**Discussion by theiamdude** [firefly-iii/firefly-iii#5049] [Firefly III]
I have just setup Firefly III using the element available on heroku (https://elements.heroku.com/buttons/firefly-iii/firefly-iii). This ends up installing the Apha version recently released (Version 5.6.0-alpha.2).

When trying use the CSV importer, I am not able to generate a token or a client ID as the PAT token is not there and the OAuth tab does not load or trigger any URL.

Is this a limitation of using Heroku? or could it be a bug with the latest version?

**Comment by JC5** [comment]
This is a Heroku limitation I'm afraid. Since it has an emphereal read-only filesystem it's not possible for Firefly III to generate the necessary keys. Everything works, *except* the oAuth keys.

**Reply by theiamdude** [reply]
Thank @JC5 for the quick reply :) is there anyway to import transactions from a CSV in a heroku setup?

**Reply by JC5** [reply]
I'm afraid not.

**Reply by theiamdude** [reply]
I am access the CLI through the console, would it possible to generate token required to use the api from there?

**Reply by JC5** [reply]
No, because the same file system problem is in the way :(
